deploy/synology_dsm.sh

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env sh
+
+# Here is a script to deploy cert to Synology DSM
+#
+# it requires the jq and curl are in the $PATH and the following
+# environment variables must be set:
+#
+# SYNO_Username - Synology Username to login (must be an administrator)
+# SYNO_Password - Synology Password to login
+# SYNO_Certificate - Certificate description to target for replacement
+#
+# The following environmental variables may be set if you don't like their
+# default values:
+#
+# SYNO_Scheme - defaults to http
+# SYNO_Hostname - defaults to localhost
+# SYNO_Port - defaults to 5000
+#
+#returns 0 means success, otherwise error.
+
+########  Public functions #####################
+
+#domain keyfile certfile cafile fullchain
+synology_dsm_deploy() {
+
+  _cdomain="$1"
+  _ckey="$2"
+  _ccert="$3"
+  _cca="$4"
+
+  _debug _cdomain "$_cdomain"
+
+  # Get Username and Password, but don't save until we successfully authenticate
+  SYNO_Username="${SYNO_Username:-$(_readdomainconf SYNO_Username)}"
```

Comment:
please just use `_getdeployconf` and `_savedeployconf`.
Don't use `xxxdomainconf()` or `xxxaccountconf()`.

see examples:

https://github.com/acmesh-official/acme.sh/blob/master/deploy/docker.sh

Reply:
Happy to change this, I was using _readdomainconf and _savedomainconf based on [your previous request](https://github.com/acmesh-official/acme.sh/pull/2369#issuecomment-520194731).